docs/fflags.adoc

```diff
@@ -0,0 +1,16 @@
+== Feature Flags
+
+The following system properties are available for toggling or specifying particular features to turn on or off.
+
+* `com.cloudbees.plugins.credentials.UseOwnPermission` - Enables the Credentials/UseOwn permission type which allows a
+user to provide their own user-scoped credentials to a build. When this feature flag is not specified, this permission
+is ignored in favor of Job/Build. When the flag is enabled, Credentials/UseOwn is only implied by Overall/Administer by
+default.
+
+* `com.cloudbees.plugins.credentials.UseItemPermission` - Enables the Credentials/UseItem permission type which allows a
```

Comment:
A guide to what “use own” and “use item” are actually _for_ would be appreciated. I for one know they exist, and could with difficulty reverse engineer the source code to determine their effect in some situation, but have no idea how you would leverage them securely to accomplish something.

Reply:
The UseItem permission is fairly simple: by adding this permission, now you can no longer access credentials from the folder or similar when running a build. For UseOwn, that is fairly difficult to describe use cases for without diving into user-scoped credentials in general which are a little anemic still. I do agree, though.

Reply:
Nothing involving `authorize-project` is “simple”. :smile:

Reply:
Alright, I played around with the feature a little bit and figured out the secret to why it's useful. The main use case is with project-based matrix authorization strategies and similar derivations. Authorize Project can add another dimension to the whole idea, but it's not required for this to be useful. It's a little convoluted, but I think you can basically use the UseItem permission to selectively limit which users can even access credentials that the job (or context) can access itself. If you get that, then you can imagine the flexibility.